### PR TITLE
[WIP] Update docs about writing plugin meta.yaml.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
# What does this PR do?
Update docs about writing plugin meta.yaml. Add info how to start remote plugin endpoint in the sidecar container.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13387

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

**Review by themr0c — COMMENTED** [review]
This is very nice!

There are still some asciidoc formatting errors to fix. For a starter:

```
asciidoctor: WARNING: proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc: line 157: no callout found for <49>
```
And all titles except the first one should be level 2 titles 
```
## title
```

```
asciidoctor: ERROR: proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc: line 161: level 0 sections can only be used when doctype is book
asciidoctor: ERROR: proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc: line 167: level 0 sections can only be used when doctype is book
asciidoctor: ERROR: proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc: line 181: level 0 sections can only be used when doctype is book
asciidoctor: ERROR: proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc: line 190: level 0 sections can only be used when doctype is book
asciidoctor: ERROR: proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc: line 276: level 0 sections can only be used when doctype is book
asciidoctor: ERROR: proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc: line 347: level 0 sections can only be used when doctype is book
```

**Comment by AndrienkoAleksandr** [comment]
@themr0c  I handled your feedback.

File: src/main/pages/che-7/end-user-guide/proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc [hunk 1]
```diff
@@ -24,11 +23,79 @@ description:                                                     <8>
 icon: https://www.eclipse.org/che/images/logo-eclipseche.svg     <9>
 repository:                                                     <10>
 category:                                                       <11>
+firstPublicationDate:                                           <12>
+latestUpdateDate:                                               <13>
+deprecate:                                                      <14>
+    autoMigrate:                                                <15>
```
Comment (olexii4) [inline src/main/pages/che-7/end-user-guide/proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc:29]: If I am not mistaken, it was implemented as 
```
deprecate {
    "automigrate": ...
    "migrateTo": ...
},
```

Comment (AndrienkoAleksandr) [reply src/main/pages/che-7/end-user-guide/proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc:29]: Than we have a bug in the spec description: https://github.com/eclipse/che-plugin-registry/blame/2b5d8a85c0d1aaf23f8a1939625a44c26c9ede55/README.md#L92

**Comment by AndrienkoAleksandr** [comment]
@rkratky could you review this pr, please?

File: src/main/pages/che-7/end-user-guide/proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc (hunk 1 above)
Comment (olexii4) [reply src/main/pages/che-7/end-user-guide/proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc:29]: Yes. Exactly. It is a new issue

**Comment by rkratky** [comment]
I'm working on a review PR for this.

**Comment by AndrienkoAleksandr** [comment]
Sorry, any news?

**Comment by rkratky** [comment]
Apologies, @AndrienkoAleksandr, this wasn't a priority for the CRW release, so it was put on the backburner. We'll be getting back to this now.

**Review by olexii4 — APPROVED** [review]

**Review by themr0c — CHANGES_REQUESTED** [review]
There is a lot of occurences of `aandrienko` in the text that seems to be some remains of a wild text replacement in the text. Can you check?

**Review by themr0c — CHANGES_REQUESTED** [review]
Here are some language improvements hints (by write-good):

```
./node_modules/.bin/write-good src/main/pages/che-7/end-user-guide/proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc 
In src/main/pages/che-7/end-user-guide/proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc

plug-in meta information is required to publish a VS Code extension in an Eclips
                         ^^^^^^^^^^^
"is required" may be passive voice on line 4 at column 29
-------------
y. This meta information is provided as a `meta.yaml` file. This section describ
                         ^^^^^^^^^^^
"is provided" may be passive voice on line 4 at column 130
-------------
<5> Type of the Che plug-in. For VS Code extensions, it must be `VS Code extensi
    ^^^^^^^
"Type of" is wordy or unneeded on line 106 at column 4
-------------
olean. If true, endpoint is exposed publicly.
                         ^^^^^^^^^^
"is exposed" may be passive voice on line 120 at column 32
-------------
If false, no k8s service is created for this endpoint.
                         ^^^^^^^^^^
"is created" may be passive voice on line 125 at column 39
-------------
olean. If true, endpoint is exposed through JWTProxy.
                         ^^^^^^^^^^
"is exposed" may be passive voice on line 126 at column 32
-------------
<26> Endpoint type. There are some special endpoint types for Eclipse Che.
                    ^^^^^^^^^
"There are" is unnecessary verbiage on line 127 at column 20
-------------
Code extension' supports only one container.
                         ^^^^
"only" can weaken meaning on line 134 at column 103
-------------
o sidecar container will be started. Example: `eclipse/che-theia-endpoint-runtim
                         ^^^^^^^^^^
"be started" may be passive voice on line 135 at column 63
-------------
<31> The RAM which is given faandrienkosidecar container by default. Example: "2
                   ^^^^^^^^
"is given" may be passive voice on line 137 at column 19
-------------
 and volume content will be erased after Eclipse Che workspace restart.
                         ^^^^^^^^^
"be erased" may be passive voice on line 146 at column 71
-------------
If true volume will be persisted and save content after Eclipse Che workspace re
                    ^^^^^^^^^^^^
"be persisted" may be passive voice on line 147 at column 20
-------------
hen you need to apply or modify some data in your containers in runtime.
                         ^^^^^^
"modify" is wordy or unneeded on line 156 at column 79
-------------
eia plug-in` that should be run in this sidecar container.
                         ^^^^^^
"be run" may be passive voice on line 158 at column 104
-------------
 container must complete successfully before the next one starts.
                         ^^^^^^^^^^^^
"successfully" can weaken meaning on line 163 at column 34
-------------
So, main goal remote plug-in endpoint - creation plugin api proxy between your r
^^
"So" adds no meaning on line 170 at column 0
-------------
 container from shell is very easy:
                         ^^^^
"very" is a weasel word and can weaken meaning on line 176 at column 63
-------------
There are a lot of images which were already created and you want to use them to
^^^^^^^^^
"There are" is unnecessary verbiage on line 277 at column 0
-------------
script pattern, than you are interested in saving call entypoint.sh. That’s why 
                         ^^^^^^^^^^^^^^
"are interested" may be passive voice on line 313 at column 62
-------------
ection. You can override only `args`:
                         ^^^^
"only" can weaken meaning on line 313 at column 203
````

**Comment by themr0c** [comment]
Here are some other language improvement hints (by proselint:

```
proselint src/main/pages/che-7/end-user-guide/proc_writing-a-meta-yaml-file-and-adding-it-to-a-plug-in-registry.adoc  --compact
-:132:76: misc.but No paragraph should start with a 'But'.
-:176:64: weasel_words.very Substitute 'damn' every time you're inclined to write 'very'; your editor will delete it and the writing will be just as it should be.
-:246:20: garner.preferred_forms 'né' is the preferred form.
```
Forget about the last line: it's in a code block and completely legit.

**Comment by AndrienkoAleksandr** [comment]
I see, docs was completed reworked. So I close this pr.

**Pull request closed without merging**